mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -694,37 +808,53 @@ def validate_columns(self, columns: set[str]) -> None:
     def __call__(
         self,
         *,
-        inputs: dict[str, Any],
-        outputs: Any,
+        inputs: dict[str, Any] | None = None,
+        outputs: Any | None = None,
         expectations: dict[str, Any] | None = None,
+        trace: Trace | None = None,
     ) -> Feedback:
         """
         Evaluate adherence to specified guidelines.
 
+        This scorer can be used in two ways:
+        1. **With a trace**: Pass an MLflow trace object to automatically extract
+           and evaluate inputs, outputs, and expectations from the trace's execution.
+        2. **With explicit parameters**: Directly provide the inputs, outputs, and expectations
+           to evaluate. This approach is more stable and predictable than trace extraction.
```

Comment:
```suggestion
        1. Pass an MLflow trace object to automatically extract
           and evaluate inputs, outputs, and expectations from the trace's execution.
        2. Directly provide the inputs, outputs, and expectations
           to evaluate. This approach is more stable and predictable than trace extraction.
```

Nit - `explicit parameters` wasn't super clear / intuitive to me. Can we remove this terminology from the PR?